R/FilterStates.R

```diff
@@ -271,7 +271,7 @@ FilterStates <- R6::R6Class( # nolint
         }
 
         # Drop teal_slices that refer to excluded variables.
-        varnames <- slices_field(state, "varname")
+        varnames <- unique(unlist(lapply(state, function(x) x[["varname"]])))
```

Comment:
FYI @chlebowa I think lapply is fine as not all `teal_slice` objects have `"varname"`. This means we can't use `vapply` in all cases

Reply:
[My point exactly.](https://github.com/insightsengineering/teal.slice/issues/381#issuecomment-1640176164)